jsonapi.go

```diff
@@ -322,6 +322,35 @@ func (d *document) verifyFullLinkage(aliasRelationships bool) error {
 	return nil
 }
 
+// verifyResourceUniqueness checks if the given document contains unique resources as described
+// by https://jsonapi.org/format/1.1/#document-resource-object-identification. Resource objects
+// across primary data and included must be unique, and resource linkages must be unique in
+// any given relationship section.
+func (d *document) verifyResourceUniqueness() bool {
+	topLevelSeen := make(map[string]struct{})
```

Comment:
Efficiency aside I think it'd be more readable for these to just be `map[string]bool` to represent the sets